tests/python/tenseal/tensors/test_bfv_tensor.py

```diff
@@ -0,0 +1,771 @@
+import copy
+import pickle
+import math
+import pytest
+
+import numpy as np
+from skimage.util.shape import view_as_windows
+
+import tenseal as ts
+
+
+SHAPES = [
+    [1],
+    [2],
+    [10],
+    [2, 2],
+    [3, 5],
+    [2, 3, 4],
+    [2, 3, 4, 5],
+    [2, 3, 4, 5, 6],
+]
+
+BROADCAST_SHAPES = [
+    ([2, 3], [2, 1]),
+    ([2, 1], [2, 4]),
+    ([2, 2], [4, 2, 2]),
+    ([2, 2, 3, 4, 5], [2, 3, 4, 5]),
+]
+
+
+def _almost_equal_number(v1, v2, m_pow_ten):
```

Comment:
Similarly, `_almost_equal_number` is only for CKKS. Here you should test for the exact result. See the test_bfv_vector tests for reference.

Reply:
Yes right, thank you